### PR TITLE
Microsoft Advertising (Bing Ads) extractor documentation

### PR DESCRIPTION
**Pull request by FNj**
Public documentation for the Microsoft Advertising (Bing Ads) extractor: https://bitbucket.org/kds_consulting_team/kds-team.ex-bingads/src/master/

@hhanova @davidesner

### CODE REVIEW

File: components/extractors/marketing-sales/bing-ads/index.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+redirect_from:
+    - /extractors/marketing-sales/bing-ads/
+---
+
+* TOC
+{:toc}
+
+The Microsoft Advertising (Bing Ads) extractor supports extracting either campaign entity data or various types of reports available. In case of reports you can specify your own set of columns and primary key to use in Keboola Storage, but there are also presets of columns and appropriate primary keys available.
+
+## Prerequisites
```
Comment (FNj) [inline components/extractors/marketing-sales/bing-ads/index.md:13]: Someone with access to a Microsoft Advertising (Bing Ads) production environment should test this.

File: components/extractors/marketing-sales/bing-ads/index.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+<!-- TODO: Please check this is a correct procedure. I could not try doing this in the production environment myself. - F. Navrkal -->
+First, you must create a Developer Token. To do so, please follow [this guide in the official documentation](https://learn.microsoft.com/en-us/advertising/guides/get-started?view=bingads-13#get-developer-token).
+
+Then you must find your Bing Ads Account ID and Customer ID. Once again, please follow [this part of the official documentation](https://learn.microsoft.com/en-us/advertising/guides/get-started?view=bingads-13#get-ids). Both should be numbers such as `391827251`.
```
Comment (FNj) [inline components/extractors/marketing-sales/bing-ads/index.md:17]: Someone with access to a Microsoft Advertising (Bing Ads) production environment should test this.

**Comment by FNj** [comment]
OK, ready for review. See the comment above. @hhanova @davidesner

**Comment by hhanova** [comment]
Thanks. I'll proofread/edit it after David is finished reviewing it. (The
same goes for the Zbozi.cz extractor.)
Hanka

File: components/extractors/marketing-sales/bing-ads/index.md (hunk 2 above)
Comment (davidesner) [reply components/extractors/marketing-sales/bing-ads/index.md:17]: This is correct, it has been tested many times by our clients who us it

**Comment by davidesner** [comment]
@FNj I have few notes / change requests:
- Move the report presets and available columns https://bitbucket.org/kds_consulting_team/kds-team.ex-bingads/src/master/docs/ to a subpage in this documentation. So that there's a table of contents and users can use links to navigate to desired section. 
  - Change all the references in text to this new subpage.
  - List the columns/metrics in `code block` element not just code. e.g. use ``` syntax

**Comment by FNj** [comment]
> @FNj I have few notes / change requests:
> 
>     * Move the report presets and available columns https://bitbucket.org/kds_consulting_team/kds-team.ex-bingads/src/master/docs/ to a subpage in this documentation. So that there's a table of contents and users can use links to navigate to desired section.
>       
>       * Change all the references in text to this new subpage.

After some thought, I don't think that is a good idea. These documents are automatically generated from the Python module defining the presets. Should the presets change ([which is already happening](https://bitbucket.org/kds_consulting_team/kds-team.ex-bingads/commits/branch/feature%2FKCOFAC-1553-bingads-extractor-v2---add-business-names-to-dimensions)), then the developer would need to not only regenerate the docs using the script, but also change them here. I think this should either be automated or it should be left as is (i.e. a link to the bitbucket repository).

Of course, if you insist, I will do this, but I would strongly advise against it.

>       List the columns/metrics in code block element not just code. e.g. use ``` syntax

This I have changed in the extractor repository.

**Comment by davidesner** [comment]
> 

I insist. Unfortunately we still don't have a mechanism in place to sync readme in component repo to here so all changes need to be done manually. Even if we add new functionality it needs to be manually duplicated. And this hopefully won't happen more often. Just please wait until we merge the new update.

**Comment by FNj** [comment]
@davidesner I've added the subpages with the column listings.

**Comment by davidesner** [comment]
Moved to #465. Closing this one

**Pull request closed without merging**